api/app/controllers/api/v1/ingests_controller.rb

```diff
@@ -0,0 +1,53 @@
+module API
+  module V1
+    # Trigger ingestion processing via the API
+    class IngestsController < ApplicationController
+      before_action :set_project, only: [:create]
+      before_action :set_text, only: [:create]
+
+      resourceful! Ingestion do
+        @project&.ingestions
+        @text&.ingestions
+      end
+
+      # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      def create
+        parent = @project || @text
+        @ingestion = ::Updaters::Ingestion
+          .new(ingestion_params)
+          .update_without_save(parent.ingestions.new)
+
+        @ingestion.creator = current_user
+
+        if @project
+          @ingestion.project_id = @project.id
+          @ingestion.target_kind = "text"
+        elsif @text
+          @project = Project.find(@text.project_id)
+          @ingestion.project_id = @project.id
+          @ingestion.target_kind = "text_section"
+          @ingestion.text_section_id = params[:data][:relationships][:text_section]
+        end
+
+        if @ingestion.save
+          @ingestion.process(current_user)
+
+          render_single_resource Text.find(@ingestion.text_id), serializer: ::V1::TextSerializer
+        else
+          render_single_resource @ingestion, serializer: ::V1::IngestionSerializer
+        end
+      end
+      # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+      private
+
+      def set_project
+        @project = Project.friendly.find(params[:id], allow_nil: true)
```

Comment:
Breaking things out into multiple controllers also allows `Model.find` to raise `ActiveRecord::RecordNotFound` where it should. As is, if someone tried to trigger an ingestion through the API on a record that had been deleted, or had a typo, or any number of other things, you could get into a place where `@project` and `@text` are both nil, and the approach here doesn't really know how to deal with that.

Splitting it out in a concern with an interface that says which type it handles allows us to raise the not found exception and process things as we do in other APIs.